### PR TITLE
2024.09

### PR DESCRIPTION
**Pull request by antgonza**

### CODE REVIEW

File: qiita_pet/support_files/doc/source/processingdata/woltka_pairedend.rst [hunk 1]
```diff
@@ -0,0 +1,72 @@
+Wolka and Bowtie2 using Read Pairing Schemes
```
Comment (antgonza) [inline qiita_pet/support_files/doc/source/processingdata/woltka_pairedend.rst:1]: @qiyunzhu; could you take a look? Thank you.

File: qiita_pet/support_files/doc/source/processingdata/processing-recommendations.rst [hunk 2]
```diff
@@ -71,17 +71,22 @@ References:
...
-Qiita currently has one active shotgun metagenomics data analysis pipeline: a per sample
+Qiita currently has one active shotgun metagenomics data analysis pipeline: a per sample, paired-end
 bowtie2 alignment step with Woltka classification using either the WoLr2 (default) or RS210 databases.
 Below you will find more information about each of these options.
 
 .. note::
-   The bowtie2 settings are maximum and minimum mismatch penalties (mp=[1,1]), a
-   penalty for ambiguities (np=1; default), read and reference gap open- and
+   The bowtie2 settings are set for interleaved processing with a maximum and minimum mismatch
+   penalties (mp=[1,1]), a penalty for ambiguities (np=1; default), read and reference gap open- and
```
Comment (charles-cowart) [inline qiita_pet/support_files/doc/source/processingdata/processing-recommendations.rst:80]: "reference gap open- and" seems odd. Perhaps it should be "open-and"? or maybe the hyphen should be removed?

File: qiita_pet/support_files/doc/source/processingdata/woltka_pairedend.rst [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+============================================
+
+Benchmarks created by Qiyun Zhu (@qiyunzhu) on Aug 1, 2024.
+
+Summary
+-------
+
+Here I tested alternative read pairing schemes in the analysis of shotgun metagenomic sequencing data. Sequencing reads were aligned against a reference microbial genome database as unpaired or paired, with or without singleton and/or discordant alignments suppressed. A series of synthetic datasets were used in the analysis.
+
+The results reveal that treating reads as paired is always advantageous over unpaired. Suppressing singleton alignments further increases the accuracy of results, despite at the cost of lower mapping rate. Suppressing discordant alignments has no obvious impact on the result. Regardless of accuracy, the downstream community ecology analyses are not obviously impacted by the choice of parameters.
```
Comment (charles-cowart) [inline qiita_pet/support_files/doc/source/processingdata/woltka_pairedend.rst:11]: despite at the cost -> despite the cost

File: qiita_pet/support_files/doc/source/processingdata/woltka_pairedend.rst [hunk 4]
```diff
@@ -0,0 +1,72 @@
+Wolka and Bowtie2 using Read Pairing Schemes
+============================================
+
+Benchmarks created by Qiyun Zhu (@qiyunzhu) on Aug 1, 2024.
+
+Summary
+-------
+
+Here I tested alternative read pairing schemes in the analysis of shotgun metagenomic sequencing data. Sequencing reads were aligned against a reference microbial genome database as unpaired or paired, with or without singleton and/or discordant alignments suppressed. A series of synthetic datasets were used in the analysis.
```
Comment (charles-cowart) [inline qiita_pet/support_files/doc/source/processingdata/woltka_pairedend.rst:9]: Here I tested -> I tested

**Review by charles-cowart — APPROVED** [review]
Approved! A few text suggestions to take or leave at your discretion.

**Review by charles-cowart — APPROVED** [review]

File: qiita_pet/support_files/doc/source/processingdata/woltka_pairedend.rst [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+Five synthetic datasets were generated with 25 samples each consisting of randomly selected WoL2 genomes. CAMISIM was executed to simulate 500 Mbp of 150 bp paired-end Illumina sequencing reads (appr. 3.3 million reads) per sample. The five datasets have different taxon count and distribution patterns. The result of one of the five datasets is displayed below. It consists of 400 taxa (more than others) and therefore is presumably the most realistic. However, all five results largely shared the same pattern.
+
+The results of the five Bowtie2 parameter sets were compared using nine metrics:
+
+Three metrics that only rely on each result.
+
+- Mapping rate (%)
+- Number of taxa
+- Entropy (i.e., Shannon index, but without subsampling)
```
Comment (qiyunzhu) [inline qiita_pet/support_files/doc/source/processingdata/woltka_pairedend.rst:37]: This line (37) can be removed. Also replace "three" with "two" in line 33.

File: qiita_pet/support_files/doc/source/processingdata/woltka_pairedend.rst [hunk 6]
```diff
@@ -0,0 +1,62 @@
+Wolka and Bowtie2 using Read Pairing Schemes
+============================================
+
+Benchmarks created by Qiyun Zhu (@qiyunzhu) on Aug 1, 2024.
+
+Summary
+-------
+
+I tested alternative read pairing schemes in the analysis of shotgun metagenomic sequencing data. Sequencing reads were aligned against a reference microbial genome database as unpaired or paired, with or without singleton and/or discordant alignments suppressed. A series of synthetic datasets were used in the analysis.
```
Comment (qiyunzhu) [inline qiita_pet/support_files/doc/source/processingdata/woltka_pairedend.rst:9]: "singleton and/or discordant alignments" is now irrelevant. It needs to be removed from this paragraph and the next two.

File: qiita_pet/support_files/doc/source/processingdata/woltka_pairedend.rst [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+
+The results reveal that treating reads as paired is always advantageous over unpaired. Suppressing singleton alignments further increases the accuracy of results, despite the cost of lower mapping rate. Suppressing discordant alignments has no obvious impact on the result. Regardless of accuracy, the downstream community ecology analyses are not obviously impacted by the choice of parameters.
+
+Therefore, I recommend the general adoption of paired alignments as a standard procedure. I also endorse suppressing singleton and discordant alignments, but note the favor of further tests on whether they may reduce sensitivity with complex communities.
+
+Alignment parameters
+--------------------
+
+Sequencing data were aligned using Bowtie2 v2.5.1 in the “very sensitive” mode against the WoL2 database. They were treated as either unpaired or paired-end:
```
Comment (qiyunzhu) [inline qiita_pet/support_files/doc/source/processingdata/woltka_pairedend.rst:18]: Double quotes may be replaced with `"`?

File: qiita_pet/support_files/doc/source/processingdata/woltka_pairedend.rst [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+.. figure::  woltka_synthetic.png
+   :align:   center
+
+
+The results revealed:
+
+#. PE outperforms SE in all metrics. Most importantly, it reduces false positive rate (higher precision) while retaining mapping rate. Meanwhile, the sensitivity (recall) of identifying true taxa is not obviously compromised (note the y-axis scale).
+#. PE.NU the two additional parameters had minimum effect on the result and make the alignment step faster. This may suggest that the additional parameters are safe to use.
+
+Therefore, I would recommend adopting paired alignment in preference to unpaired alignment. I may suggest no mixing as it has improved accuracy, but the potential adverse effect of lower mapping rate may be further explored before making a compelling recommendation. Although not having a visible effect, no discordance may be added for logical coherency.
```
Comment (qiyunzhu) [inline qiita_pet/support_files/doc/source/processingdata/woltka_pairedend.rst:62]: Can remove "no mixing... no discordance".

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,5 +1,20 @@
...
+
+Deployed on September 23rd, 2024
+
+* Added update_resource_allocation_redis and companion code, so resource allocations summaries are available for review. Thank you @Gossty!
+* Now is possible to have default workflows with only one step.
+* `qiita_client.update_job_step` now accepts an ignore_error optional parameter. Thank you @charles-cowart!
+* Initial changes in `qiita_client` to have more accurate variable names: `QIITA_SERVER_CERT` -> `QIITA_ROOTCA_CERT`. Thank you @charles-cowart!
+* Added `get_artifact_html_summary` to `qiita_client` to retrieve the summary file of an artifact.
+* Re-added github actions to `https://github.com/qiita-spots/qiita_client`.
+* `Woltka v0.1.4, paired-end` superseded `Woltka v0.1.4` in `qp-woltka`; [more information](https://qiita.ucsd.edu/static/doc/html/processingdata/woltka_pairedend.html). Thank you to @qiyunzhu for the benchmarks!
```
Comment (qiyunzhu) [inline CHANGELOG.md:14]: Should upgrade Woltka to v0.1.6. (!important!)

**Comment by antgonza** [comment]
Thank you @qiyunzhu, I made the suggested changes in a new PR: https://github.com/qiita-spots/qiita/pull/3434

**Pull request merged**